tests/integration/spark_pipeline/test_deaths_per_year_pipeline/test_death_per_year.py

```diff
@@ -0,0 +1,38 @@
+import pathlib
+from unittest import mock
+
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.types import LongType, StringType, StructField, StructType
+
+from spark_pipeline import deaths_per_year_pipeline
+
+CURRENT_PATH = pathlib.Path(__file__).parent.resolve()
+
+
+@mock.patch("spark_pipeline.base.read_write.write_csv", autospec=True)
+def test_pipeline(mock_write_jdbc: mock.Mock):
+    # arrange
+    spark = SparkSession.builder.getOrCreate()
+    schema = StructType(
+        [
+            StructField("total_deaths", LongType(), True),
+            StructField("death_year", StringType(), True),
+        ]
+    )
+    expected_df = spark.read.load(
+        path=f"{CURRENT_PATH}/expected_output.csv",
+        format="csv",
+        schema=schema,
+        header="true",
+    )
+    pipeline = deaths_per_year_pipeline.DeathsPerYearPipeline(
+        input_path=f"{CURRENT_PATH}/input.csv"
+    )
+
+    # act
+    deaths_per_year_pipeline.run(pipeline)
+    output_df: DataFrame = mock_write_jdbc.call_args.kwargs["df"]
+
+    # assert
+    assert sorted(output_df.collect()) == sorted(expected_df.collect())
+    assert mock_write_jdbc.call_args.kwargs["mode"] == "overwrite"
```

Comment:
I tried to improve the organization of this integration test. I think is much cleaner now right? I always like to have these `arrange`, `act`, and `assert` stages well organized and defined, it helps in the overall readability of the test.

Reply:
What does this act stage do exactly? Is it creating a mock object?

Reply:
run the pipeline, and capture the final df that is passed to the `write_csv` service

Reply:
sorry, I have made a mistake while naming this mock object, here the fix: https://github.com/joao-victor-campos/pyspark-data-aggregation-pipeline/pull/14/commits/aa3a7928db47e34cb7acd04719ed8a0caf0f433d